conwayste/src/client.rs

```diff
@@ -1399,6 +1427,32 @@ impl MainState {
         }
     }
 
+    fn bit_pattern_from_char(&self, keycode: KeyCode) -> (BitGrid, usize, usize) {
+        // TODO: make this configurable
+        let rle_str = match keycode {
+            KeyCode::Key2 => "bob$2bo$3o!",  // SE glider
+            KeyCode::Key3 => "o$obo$2o!",    // SW glider
+            KeyCode::Key4 => "3o$o$bo!",     // NW glider
+            KeyCode::Key5 => "b2o$obo$2bo!", // NE glider
+
+            KeyCode::Key6 => "4bo$5bo$o4bo$b5o!",         // E LWSS
+            KeyCode::Key7 => "bo$o$o$o$o2bo$3o!",         // S LWSS
+            KeyCode::Key8 => "5o$o4bo$o$bo!",             // W LWSS
+            KeyCode::Key9 => "b3o$o2bo$3bo$3bo$3bo$2bo!", // N LWSS
+
+            // https://www.conwaylife.com/wiki/Period-22_glider_gun
+            KeyCode::Key0 => concat!("18b2o25b$19bo7bo17b$19bobo14b2o7b$20b2o12b2o2bo6b$24b3o7b2ob2o6b$24b2o",
+                                     "b2o7b3o6b$24bo2b2o12b2o2b$25b2o14bobob$35bo7bob$43b2o2$2o23bo19b$bo21b",
+                                     "obo19b$bobo13b3o4b2o19b$2b2o3bo8bo3bo24b$6bob2o6bo4bo23b$5bo4bo6b2obo",
+                                     "9bo14b$6bo3bo8bo3b2o6bo13b$7b3o13bobo3b3o13b$25bo19b$25b2o!"),
+            _ => "", // unexpected
+        };
+        let pat = Pattern(rle_str.to_owned());
+        let (width, height) = pat.calc_size().unwrap();  // unwrap OK because calc_size should never fail on valid RLE
```

Comment:
If it does fail then we've hardcoded an RLE incorrectly, or libconway produced an incorrect one, and we'd want to fix that prior to a release. 
:+1:

Reply:
Fixed. Now we will see something like this in the logs if the RLE is bad:
```
Wed 2019-10-30 21:30:28.226259 [ERROR] - Received unexpected error from process_running_inputs(). "Invalid pattern bound to keycode Key8: ConwayError->InvalidData->Premature termination at Some(12)"
```

The 12 is a character number.

Reply:
Beautiful!